### PR TITLE
Add project specific steps to issue and PR checklists

### LINKED ISSUE #21: Add additional steps to issue/pr checklists
**Is your feature request related to a problem? Please describe.**
There are quite a lot of project specific steps missing on the PR and issue templates that should be added.

**Describe the solution you'd like**
Checklists should be extended based off of items on [GitHub checklist](https://docs.google.com/document/d/1B5vqW28E2EWsvdM-lrHi_9xHrk4mNdftcG2VqgGOatI/edit?usp=sharing). Not all items on there will be applicable so use discretion.

Issue templates should have checklists added

### PR DESCRIPTION
**Pull request by oscarli00**
# Description

Adds extra project specific steps to the issue and PR checklists.

Fixes/resolves #21 

## Type of change
Please delete options that are not relevant.

- [x] **Improvement** (non-breaking change which improves existing functionality)
- [x] This change requires a documentation update

# Checklist:
Leave blank if not applicable

I have completed these steps when making this pull request:
- [x] I have assigned my name to the issue
- [x] I have moved the issue to the **In Progress** column
- [x] I have labelled the PR appropriately
- [x] I have assigned myself to the PR

Complete these steps as you work through the issue:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have moved the linked issue to the **Review in Progress** column

### CODE REVIEW

**Comment by j-chad** [comment]
before you open this PR for review, can you please copy the PR template into this PR description

**Comment by oscarli00** [comment]
Let me know if the checklist is too long or missing something

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -8,13 +8,24 @@ assignees: ''
...
 A clear and concise description of any alternative solutions or features you've considered.
 
 **Additional context**
+
 Add any other context or screenshots about the feature request here.
+
+### Checklist:
+
+- [ ] I have double checked that the issue does not currently exist. 
+- [ ] I have filled out the above template.
```
Comment (j-chad) [inline .github/ISSUE_TEMPLATE/feature_request.md]: I suggest we remove "I have filled out the template" from the checklist as going through the checklist means they are filling it out

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -8,31 +8,45 @@ assignees: ''
...
  - Version [e.g. 22]
 
 **Additional context**
+
 Add any other context about the problem here.
+
+### Checklist:
+
+- [ ] I have double checked that the issue does not currently exist. 
+- [ ] I have filled out the above template.
```
Comment (j-chad) [inline .github/ISSUE_TEMPLATE/bug_report.md]: I suggest we remove "I have filled out the template" from the checklist as going through the checklist means they are filling it out

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -19,10 +19,25 @@ Please delete options that are not relevant.
 # Checklist:
```
Comment (j-chad) [inline .github/pull_request_template.md:19]: I think the checklist should look like this:

# Checklist
Leave blank if not applicable

I have completed these steps when making this pull request:
- [ ] I have assigned my name to the issue
- [ ] I have moved the issue to the **In Progress** column
- [ ] I have labelled the PR appropriately

Before opening PR for review:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have moved the linked issue to the **Review in Progress** column

Comment (j-chad) [reply .github/pull_request_template.md:19]: we should also add I have assigned my name to this PR

Comment (oscarli00) [reply .github/pull_request_template.md:19]: I've made the suggested changes

**Review by j-chad — APPROVED** [review]
LGTM

**Pull request merged**